File: ruby/easy/reshape_the_matrix.rb
# @param {Integer[][]} nums
# @param {Integer} r
# @param {Integer} c
# @return {Integer[][]}
def matrix_reshape(nums, r, c)
    res, temp = [], []
    k = 0
    for i in 0..nums.length-1
        for j in 0..nums[i].length-1
            if k >= c
                res.push(temp)
                temp = []
                k = 0
            end
            temp.push(nums[i][j])
            k += 1
        end
    end
    res.push(temp)
    return (res.length == r && res[0].length == c) ? res : nums
end

File: javascript/easy/reshapeTheMatrix.js
/**
 * @param {number[][]} nums
 * @param {number} r
 * @param {number} c
 * @return {number[][]}
 */
var matrixReshape = function(nums, r, c) {
    let result = new Array(), temp = new Array(), k = 0;
    for(let i = 0; i < nums.length; i++) {
        for(let j = 0; j < nums[i].length; j++) {
            if(k >= c) {
                result.push(temp);
                temp = new Array();
                k = 0;
            }
            temp.push(nums[i][j]);
            k++;
        }
    }
    result.push(temp);
    return (result.length === r && result[0].length === c) ? result : nums;
};
